code/game/objects/items/stacks/sandbags.dm

```diff
@@ -105,8 +105,8 @@
 		return
 	user.visible_message("<span class='notice'>[user] starts assembling a sandbag barricade.</span>",
 	"<span class='notice'>You start assembling a sandbag barricade.</span>")
-
-	if(!do_after(user, 20, TRUE, src, BUSY_ICON_BUILD))
+	var/building_time = 2 SECONDS *(1 - clamp(user.skills.getRating("construction") * 0.10, 0 , 0.50))
```

Comment:
use LERP here and replace fastest speed with low as can go and maximumskill with the point it reaches the maximum speed
```suggestion
	var/building_time = LERP(2 SECONDS, fastestspeed, user.skills.getPercent("construction", maximumskill)
```